src/platform/Darwin/DiagnosticDataProviderImpl.cpp

```diff
@@ -89,6 +89,21 @@ CHIP_ERROR DiagnosticDataProviderImpl::ResetWatermarks()
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR DiagnosticDataProviderImpl::GetRebootCount(uint16_t & rebootCount)
+{
+    uint32_t count = 0;
```

Comment:
why is this uint32_t? and then cast to uint16_t on line 101?

Reply:
> why is this uint32_t? and then cast to uint16_t on line 101?

Because the configuration manager API is:
```
virtual CHIP_ERROR GetRebootCount(uint32_t & rebootCount)                          = 0;
```

Now why is that one 32-bit while the actual value in the spec is 16-bit?  I don't really know, and it's very out of scope for this PR to change.